src/CLaSH/Signal/Internal.hs

```diff
@@ -269,7 +269,7 @@ register# _ i s = i :- s
 regEn# :: SClock clk -> a -> Signal' clk Bool -> Signal' clk a -> Signal' clk a
 regEn# _ = go
   where
-    go o (e :- es) as@(~(x :- xs)) =
+    go o ~(e :- es) as@(~(x :- xs)) =
```

Comment:
Actually, you were right, I was wrong. Could you revert back to the original code and add a comment go `go` along the lines of:

"Because both the enable and in the value argument of the register can be used in a feedback loop, we need lazy pattern matching on both those arguments. That is, in order to evaluate the first output sample `o :- ...`, we don't need to know the shape of our inputs. However, the enable and value argument should be evaluated to WHNF as soon as the tail of the result is evaluated to WHNF, otherwise we have a space-leak. The value argument `as` therefor needs an explicit `seq`; on the other hand, because the tail of the output depends on the value of `e`, the enable argument does not need an explicit `seq`."